packages/style-engine/class-wp-style-engine-css-rule.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * WP_Style_Engine_CSS_Rule
+ *
+ * An object for CSS rules.
+ *
+ * @package Gutenberg
+ */
+
+if ( class_exists( 'WP_Style_Engine_CSS_Rule' ) ) {
+	return;
+}
+
+/**
+ * Holds, sanitizes, processes and prints CSS declarations for the style engine.
+ *
+ * @access private
+ */
+class WP_Style_Engine_CSS_Rule {
+
+	/**
+	 * The selector.
+	 *
+	 * @var string
+	 */
+	protected $selector;
+
+	/**
+	 * The selector declarations.
+	 *
+	 * Contains an array of WP_Style_Engine_CSS_Declarations objects.
+	 *
+	 * @var WP_Style_Engine_CSS_Declarations[]
+	 */
+	protected $declarations = array();
+
+	/**
+	 * Constructor
+	 *
+	 * @param string                             $selector     The CSS selector.
+	 * @param WP_Style_Engine_CSS_Declarations[] $declarations An array of WP_Style_Engine_CSS_Declarations objects.
+	 */
+	public function __construct( $selector = '', $declarations = array() ) {
+		$this->set_selector( $selector );
+		$this->set_declarations( $declarations );
+	}
+
+	/**
+	 * Set the selector.
+	 *
+	 * @param string $selector The CSS selector.
+	 */
+	public function set_selector( $selector ) {
+		if ( empty( $selector ) ) {
+			return;
+		}
+		$this->selector = esc_html( $selector );
+	}
+
+	/**
+	 * Set the declarations.
+	 *
+	 * @param WP_Style_Engine_CSS_Declarations[] $declarations An array of WP_Style_Engine_CSS_Declarations objects.
+	 */
+	public function set_declarations( $declarations ) {
+		if ( $declarations instanceof WP_Style_Engine_CSS_Declarations ) {
+			$declarations = array( $declarations );
+		}
+		$this->declarations = array_merge( $this->declarations, $declarations );
```

Comment:
Is it possible that we'd set new declarations after the initial instantiation? 

If the incoming declarations are instances of `WP_Style_Engine_CSS_Declarations`, how should we merge and deal with duplicate properties in the declarations?

Reply:
If the intention is to store `WP_Style_Engine_CSS_Declarations` instances only, should we have a stricter check? Example: https://github.com/WordPress/gutenberg/pull/42408/files#diff-1797702223221e71dd529efd7a02cf619e4cf5fddf1719ef5805a8c05c8ecbcdR44